### PR TITLE
Sort sobol

### LINKED ISSUE #827: Sobol output is random
--------
Issue Description
--------
##### What did you expect to see happen?
Sobol to have consistent output between python 2 and python 3.

##### What did you see instead?
The output varied.

##### Do you have a suggested fix for the development team?
Stop using unsorted .items() from a dictionary.

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.
framework/Samplers/Sobol.Sobol

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [ ] 1. Is it tagged with a type: defect or task?
- [ ] 2. Is it tagged with a priority: critical, normal or minor?
- [ ] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [ ] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [ ] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address? 
Addresses #827, but further investigation needed to close. 

##### What are the significant changes in functionality due to this change request?
Sorts the output from the Sobol sampler to get consistent results.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: tests/framework/PostProcessors/InterfacedPostProcessor/gold/hs_sync_all/synced.csv [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
-1.11270166538,0.0,synced_1.csv
-1.88729833462,0.0,synced_2.csv
-1.5,-0.774596669241,synced_3.csv
-1.5,0.774596669241,synced_4.csv
+x,y,filename
+0.0,1.5,synced_0.csv
+-0.774596669241,1.5,synced_1.csv
+0.774596669241,1.5,synced_2.csv
+0.0,1.11270166538,synced_3.csv
+0.0,1.88729833462,synced_4.csv
```
Comment (PaulTalbot-INL) [inline tests/framework/PostProcessors/InterfacedPostProcessor/gold/hs_sync_all/synced.csv]: The "synced" files should be following the order given by the DataSet, so the sorting of the ROMs shouldn't have changed this, right? The order of the outputs should be controlled by the DataSet that they're put in. If not, we probably need to figure out why. Did changing the line in the Sobol control the variable order?

Comment (joshua-cogliati-inl) [reply tests/framework/PostProcessors/InterfacedPostProcessor/gold/hs_sync_all/synced.csv]: Based on my reading of test_historySetSyncAll.xml, the dataset is generated in the make step, and then synced in the print step, and this changes the dataset at the make step so it is different when it gets to the hs_sync_all/synced.csv  later.    Yes, it controls the variable order (the order of the ROMs).

Comment (PaulTalbot-INL) [reply tests/framework/PostProcessors/InterfacedPostProcessor/gold/hs_sync_all/synced.csv]: Okay, I will just open an issue to investigate that then. The variables should be entirely in the order of how they're listed in the input file I think. In the meantime, these changes are fine to me.

**Comment by PaulTalbot-INL** [comment]
Checklist passes.

Qsub tests currently disabled, all required tests passing.

Modifying to avoid closing the linked issue, and merging.

**Pull request merged**